src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -0,0 +1,352 @@
+import json
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+from click import ClickException
+from snowflake.cli.api.console import cli_console as cc
+from snowflake.cli.api.project.schemas.native_app.native_app import NativeApp
+from snowflake.cli.api.project.schemas.native_app.path_mapping import (
+    PathMapping,
+    Processor,
+)
+from snowflake.cli.plugins.nativeapp.artifacts import (
+    is_glob,
+    resolve_without_follow,
+)
+from snowflake.cli.plugins.nativeapp.codegen.snowpark.extension_function import (
+    ExtensionFunctionProperties,
+)
+from snowflake.cli.plugins.nativeapp.codegen.snowpark.processor_base import (
+    BuiltInAnnotationProcessor,
+)
+from snowflake.cli.plugins.nativeapp.utils import (
+    get_all_file_paths_under_dir,
+    get_all_python_files,
+)
+
+
+class VirtualEnvironmentType(Enum):
+    AUTO_DETECT = 0
+    VENV = 1
+    CONDA = 2
+    SYSTEM_DEFAULT = 3
+
+
+def determine_virtual_env(processor: Processor) -> Dict[str, Any]:
+    """
+    Determines a virtual environment to run the Snowpark processor in, either through the project definition or by querying the current environment.
+    """
+    if "env" in processor.properties:
+        env_type = processor.properties["env"]["type"]
+        if env_type.lower() == "conda":
+            if "name" not in processor.properties["env"]:
+                cc.warning(
+                    "Please provide a name for the conda environment to run the Snowpark processor in. Will auto-detect and use the currently active environment."
+                )
+            else:
+                return {
+                    "env_type": VirtualEnvironmentType.CONDA,
+                    "name": processor.properties["env"]["name"],
+                }
+        elif env_type.lower() == "venv":
+            if "path" not in processor.properties["env"]:
+                cc.warning(
+                    "Please provide a path for the venv to run the Snowpark processor in. Will auto-detect and use the currently active environment."
+                )
+            else:
+                return {
+                    "env_type": VirtualEnvironmentType.VENV,
+                    "path": processor.properties["env"]["path"],
+                }
+    return {}
+
+
+def execute_in_sandbox(py_file: str, kwargs: Dict[str, Any]):
+    # Create the code snippet to be executed in the sandbox
+
+    script_source = f"""
+import functools
+import sys
+
+try:
+    import snowflake.snowpark
+except ModuleNotFoundError:
+    sys.exit(1)
```

Comment:
Actually, do we even need to surround this in try except? Because if the import fails due to `ModuleNotFoundError`, that will abort the rest of the execution, and the exception should bubble up to the subprocess.stderr. Is that an incorrect understanding?